pkg/asset/manifests/network.go

```diff
@@ -76,6 +77,13 @@ func (no *Networking) Generate(dependencies asset.Parents) error {
 		serviceNet = append(serviceNet, sn.String())
 	}
 
+	networkType := netConfig.NetworkType
+	// For backward-compatibility reasons the installer needs to accept "OpenShiftSDN" in the
+	// wrong case, but other components aren't expected to deal with it.
+	if strings.ToLower(networkType) == "openshiftsdn" {
+		networkType = "OpenShiftSDN"
```

Comment:
And also maybe `OVNKubernetes` and `Kuryr`?  I was [expecting][1] this canonicalization to use [this existing variable][2], which we could make public and move around if needed, and also [the networking conversion handling][2], so we ensured that everyone consuming the install-config would have the canonical casing.

[1]: https://github.com/openshift/machine-config-operator/pull/2101#discussion_r491061293
[2]: https://github.com/openshift/installer/blob/79510119c59212ed1021a4bf96233df646667b2e/pkg/types/validation/installconfig.go#L42
[3]: https://github.com/openshift/installer/blob/79510119c59212ed1021a4bf96233df646667b2e/pkg/types/conversion/installconfig.go#L38

Reply:
> And also maybe `OVNKubernetes` and `Kuryr`?

The original bug was only about accepting both `OpenshiftSDN` and `OpenShiftSDN`. The fix was implemented by doing a case-insensitive match, so I did the same here since theoretically someone might be doing `networkType: openshiftsdn` and it currently works...

CNO currently also accepts `OVNKubernetes` case-insensitively even though there is no historical reason for that. Given that ovn-kubernetes was not GA until now, and that we don't do case-insensitive matching on any other plugin names (including `Kuryr`), I think the fix for this is to make CNO stop accepting `OVNKubernetes` in the wrong case.

> I was [expecting](https://github.com/openshift/machine-config-operator/pull/2101#discussion_r491061293) this canonicalization to use [this existing variable](https://github.com/openshift/installer/blob/79510119c59212ed1021a4bf96233df646667b2e/pkg/types/validation/installconfig.go#L42), which we could make public and move around if needed

Hm... I did a grep and saw that the string `"OpenShiftSDN"` was already hardcoded in multiple places, but looking more carefully, most of those places are unit tests. But it seems like really it ought to be defined in openshift/api ? Would you be OK with leaving it hardcoded now but fixing it to use consts from api in 4.7?

> and also [the networking conversion handling](https://github.com/openshift/installer/blob/79510119c59212ed1021a4bf96233df646667b2e/pkg/types/validation/installconfig.go#L42)

wrong link (it's the same link as the previous one)

Reply:
oh wait, they are defined in openshift/api, it's just in openshift/api/operator, not openshift/api/config

Reply:
Oops.  Networking conversion: https://github.com/openshift/installer/blob/79510119c59212ed1021a4bf96233df646667b2e/pkg/types/conversion/installconfig.go#L39
I'm fine restricting canonicalization to only values where network tooling has canonicalization logic, regardless of how that combines other installer adding canonicalization and networking removing canonicalization.